src/main/java/com/faforever/api/leaderboard/GlobalLeaderboardRepository.java

```diff
@@ -23,14 +23,40 @@
     "     WHERE (expires_at is null or expires_at > NOW()) AND revoke_time IS NULL" +
     "  ) " +
     "  ORDER BY rating DESC LIMIT ?#{#pageable.offset},?#{#pageable.pageSize}",
-    countQuery = "SELECT count(*) FROM ladder1v1_rating WHERE is_active = 1 AND ladder1v1_rating.numGames > 0" +
+    countQuery = "SELECT count(*) FROM global_rating WHERE is_active = 1 AND global_rating.numGames > 0" +
       "   AND id NOT IN (" +
       "     SELECT player_id FROM ban" +
       "     WHERE (expires_at is null or expires_at > NOW()) AND revoke_time IS NULL" +
       "       AND (1=1 OR -1 IN (?,?,?))" +
       "  ) -- Dummy placeholder for pageable, prevents 'Unknown parameter position': ?,?,?", nativeQuery = true)
   Page<GlobalLeaderboardEntry> getLeaderboardByPage(Pageable pageable);
 
+  @Query(value = "SELECT" +
+    "    global_rating.id," +
+    "    login.login," +
+    "    global_rating.mean," +
+    "    global_rating.deviation," +
+    "    global_rating.numGames," +
+    "    @s \\:= @s + 1 rank" +
+    "  FROM global_rating JOIN login on login.id = global_rating.id," +
+    "    (SELECT @s \\:= ?#{#pageable.offset}) AS s" +
+    "  WHERE is_active = 1 AND global_rating.numGames > 0" +
+    "   AND login.id NOT IN (" +
+    "     SELECT player_id FROM ban" +
+    "     WHERE (expires_at is null or expires_at > NOW()) AND revoke_time IS NULL" +
+    "  ) " +
+    "   AND login.login LIKE ?#{#playerNameRegex}"+
```

Comment:
now that I think of, is this really what we want to achieve? original query relies on all players to be taken into account to set proper rank. now if you filter by name, you will rank from 1 all matching players, which I suppose is not the goal.
Overall shouldn't we change this to materialized view refreshed every few minutes and use regular elide mapping?
@Brutus5000 @norraxx ?

Reply:
U are right there

Reply:
Not sure if @Brutus5000 is ok with that

Reply:
So I would know how to correct it... just make the +1 part in a select

Reply:
We have a scheduled job already to purge the leaderboards for inactive players. We could as well make a scheduled job to rebuild ranked leaderboards each time (using MySQL events it is the closest to a materialized view we can get)

If we can do that, we can pull it into Elide.

Reply:
🔝 will try one day